airflow_prometheus_exporter/prometheus_exporter.py

```diff
@@ -450,9 +450,13 @@ def collect(self):
             labels=["task_id", "dag_id", "execution_date"],
         )
         for task in get_landing_times():
-            task_duration_value = (
-                task.end_date - date_range(task.execution_date, num=2, delta=task.schedule_interval)[1]
-            ).total_seconds()
+            if task.schedule_interval:
+                task_duration_value = (
+                    task.end_date - date_range(task.execution_date, num=2, delta=task.schedule_interval)[1]
+                ).total_seconds()
+            else:
+                task_duration_value = (task.end_date - task.execution_date).total_seconds()
```

Comment:
What guarantees you these are not none either?

Reply:
[This line](https://github.com/jampp/airflow-prometheus-exporter/blob/develop/airflow_prometheus_exporter/prometheus_exporter.py#L299
) guaranties end_date.

If the task ended, then it has execution date.